Content.Shared/_Impstation/Clothing/Obvious/WearerGetsExamineTextSystem.cs

```diff
@@ -0,0 +1,113 @@
+/// </summary>
+public sealed class WearerGetsExamineTextSystem : EntitySystem
+{
+    [Dependency] private readonly IEntityManager _entManager = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<WearerGetsExamineTextComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<WearerGetsExamineTextComponent, GotUnequippedEvent>(OnUnequipped);
+        SubscribeLocalEvent<WearerGetsExamineTextComponent, ExaminedEvent>(OnExamine);
+    }
+
+    private void OnEquipped(Entity<WearerGetsExamineTextComponent> entity, ref GotEquippedEvent args)
+    {
+        if (!TryComp(entity, out ClothingComponent? clothing))
+            return;
+        var isCorrectSlot = (clothing.Slots & args.SlotFlags) != Inventory.SlotFlags.NONE;
+        if (!entity.Comp.PocketEvident) //if it can't be evident in our pockets
+        {
+            // Make sure the clothing item was equipped to the right slot, and not just held in a hand.
+            if (!isCorrectSlot)
+                return;
+        }
+
+        entity.Comp.Wearer = args.Equipee;
+        Dirty(entity);
+
+        //GIVE THEM INSPECT TEXT
+        var obviousExamine = EnsureComp<ExtraExamineTextComponent>(args.Equipee);
+        obviousExamine.Lines.Remove(entity.Comp.ExamineOnWearer); // if prefix needs to be updated
+        obviousExamine.Lines.TryAdd(entity.Comp.ExamineOnWearer,  //using try so that we don't cause an error if we move something from slot to slot
+            isCorrectSlot ? entity.Comp.PrefixExamineOnWearer : "obvious-prefix-have"); // uses a different prefix if worn / displayed
+    }
+
+    private void OnUnequipped(Entity<WearerGetsExamineTextComponent> entity, ref GotUnequippedEvent args)
+    {
+        //REMOVE THE FUCKIN INSPECT TEXT
+        if (entity.Comp.Wearer == null)
+            return;
+
+        var wearer = entity.Comp.Wearer.Value;
+        if (TryComp(wearer, out ExtraExamineTextComponent? obviousExamine))
+        {
+            obviousExamine.Lines.Remove(entity.Comp.ExamineOnWearer);
+        }
+
+        entity.Comp.Wearer = null;
+        Dirty(entity);
+    }
+
+    private void OnExamine(Entity<WearerGetsExamineTextComponent> entity, ref ExaminedEvent args)
+    {
+        var currentlyWorn = entity.Comp.Wearer != null;
+        var contraString = "";
+        var outString = "";
+        outString = Loc.GetString(currentlyWorn ? "obvious-on-item-currently" : "obvious-on-item",
+            ("used", Loc.GetString(entity.Comp.PocketEvident ? "obvious-type-pockets" : "obvious-type-default")),
+            ("thing", entity.Comp.Thing),
+            ("me", Identity.Entity(entity, EntityManager)));
+        if (entity.Comp.WarnExamine && !currentlyWorn && TryComp(entity, out ContrabandComponent? contra))
+        {
+            switch (contra.Severity) // apply additional text if the item is contraband to note that displaying it might be really bad
```

Comment:
might be worth just appending the contraband severity to the end of "obvious-on-item-contra-" so this doesn't have to be manually expanded in the case of additional contraband severities